src/acl.c

```diff
@@ -1573,6 +1582,15 @@ void cleanupACLKeyResultCache(aclKeyResultCache *cache) {
     if (cache->keys_init) getKeysFreeResult(&(cache->keys));
 }
 
+/* Fetch the keys from the command to be used for ACLs. */
+int ACLGetKeysFromCommand(struct redisCommand *cmd, robj **argv, int argc, int search_flags, getKeysResult *result) {
+    /* In case of sort and sort_ro we want to get the command access patterns used by 'get' and 'by' */
+    if (cmd->proc == sortCommand || cmd->proc == sortroCommand)
+        return sortGetKeysWithPatterns(cmd, argv, argc, result);
+    else
+        return getKeysFromCommandWithSpecs(cmd, argv, argc, search_flags, result);
```

Comment:
I will chime in that I didn't suggest a generic mechanism since it was specific to just ACLs. Adding generic mechanisms convolutes code and makes you look at different levels of abstraction to understand the functionality, which seems unwarranted here.

Reply:
I was playing with the idea to add a flag to commands which accept patterns as argument, and make the generic ACL getKeysFromCommandWithSpecs return patterns in case query is from ACL. I just wanted to reduce the change to be less intrusive. However I  am not sure I understand your suggestion in this case @oranagra

Reply:
i was not suggesting to add a command flag, i was arguing that this needs to be an explicit check done in `sortCommand`, instead of a "generic" mechanism that detects it's acting on behalf of SORT and behaves differently.

I'd like to respond here, also to your other reply saying:
> but future commands which might except access patterns would also need to have a specialized handling of ACL matching

we will never make that SORT mistake again, it's a mistake that AFAIK Salvatore regretted soon after adding, which gave rise to Lua scripts.
we keep that just for backwards compatibility, and we'll never add another command that accesses keys by using key names extracted from value of other keys, be it depending on a pattern given by an argument, or otherwise.

so the point is, that it doesn't need to be a generic infrastructure.